pkgs/development/libraries/qt-5/hooks/qtbase-setup-hook.sh

```diff
@@ -62,3 +62,59 @@ postPatchMkspecs() {
 if [ -z "$dontPatchMkspecs" ]; then
     postPhases="${postPhases}${postPhases:+ }postPatchMkspecs"
 fi
+
+_Qt_sortless_uniq() {
+	# `uniq`, but keeps initial order.
+	# This is to remove risks of combinatorial explosion of plugin paths.
+	cat -n | sort -uk2 | sort -nk1 | cut -f2-
+}
+
+_QtGetPluginPaths() {
+	# Lists all plugin paths for current Qt for given buildInputs and propagatedBuildInputs
+	local i
+	local _i
+	local o
+	local inputs
+
+	# FIXME : this causes output path cycles...
+	# I am unsure if it is even needed, though.
```

Comment:
Oh, and there's this...

I'm not sure I need to do this, but maybe I need, otherwise some parts may be missing.

The situation I'm thinking about is just like how `qtbase` has `dev` and `bin`. When you `buildInputs = [ qt5.qtbase ]` you get `qt5.qtbase.dev`. This is fine, except the plugins are in `qt5.qtbase.bin`, which won't get figured out. This is why I'm forcibly adding it to `nix-support/qt-plugin-paths` at build time.

I'll have to first wrap my mind around what the error is, then figure out something.

Reply:
> The situation I'm thinking about is just like how `qtbase` has `dev` and `bin`.

This is the situation for almost all libraries. I noticed that you hardcoded the dependency on `qtbase` below; wouldn't we need to do this for all Qt libraries? (Missing the plugins from `qtbase` is the usually the only problem that is fatal, but missing the plugins from other libraries still leads to strange behavior and missing features.)